app.js

```diff
@@ -1 +1,165 @@
 'use stict';
+
+// CREATE AN ARRAY OF ALL THE POSSIBLE IMAGES
+var allImages = [];
+// 0
+new FocusImage('R2D2 bag', 'img/bag.jpg', 'bag-img', allImages);
+// 1
+new FocusImage('banana slicer', 'img/banana.jpg', 'banana-img', allImages);
+// 2
+new FocusImage('bathroom tablet holder', 'img/bathroom.jpg', 'bathroom-img', allImages);
+// 3
+new FocusImage('toeless boots', 'img/boots.jpg', 'boots-img', allImages);
+// 4
+new FocusImage('all-in-one breakfast machine', 'img/breakfast.jpg', 'breakfast-img', allImages);
+// 5
+new FocusImage('meatball bubblegum', 'img/bubblegum.jpg', 'bubblegum-img', allImages);
+// 6
+new FocusImage('inverted chair', 'img/chair.jpg', 'chair-img', allImages);
+// 7
+new FocusImage('chthlhu action figure', 'img/cthulhu.jpg', 'cthulhu-img', allImages);
+// 8
+new FocusImage('dog duck beak', 'img/dog-duck.jpg', 'dog-duck-img', allImages);
+// 9
+new FocusImage('dragon meat', 'img/dragon.jpg', 'dragon-img', allImages);
+// 10
+new FocusImage('utensil pen caps', 'img/pen.jpg', 'pen-img', allImages);
+// 11
+new FocusImage('pet footie sweepers', 'img/pet-sweep.jpg', 'pet-sweep-img', allImages);
+// 12
+new FocusImage('pizza scissors', 'img/scissors.jpg', 'scissors-img', allImages);
+// 13
+new FocusImage('shark sleeping bag', 'img/shark.jpg', 'shark-img', allImages);
+// 14
+new FocusImage('baby onesie sweeper', 'img/sweep.png', 'sweep-img', allImages);
+// 15
+new FocusImage('tauntaun sleeping bag', 'img/tauntaun.jpg', 'tauntaun-img', allImages);
+// 16
+new FocusImage('unicorn meat', 'img/unicorn.jpg', 'unicorn-img', allImages);
+// 17
+new FocusImage('tentacle usb', 'img/usb.gif', 'usb-img', allImages);
+// 18
+new FocusImage('self-watering can', 'img/water-can.jpg', 'water-can-img', allImages);
+// 1
+new FocusImage('closed-top wine glass', 'img/wine-glass.jpg', 'wine-glass-img', allImages);
+
+/**
+creates an object for a image with the name, path, and id. adds to given array.
+**/
+function FocusImage(imageTitle, imagePath, imageId, imageArray) {
+  this.imageTitle = imageTitle;
+  this.imagePath = imagePath;
+  this.imageId = imageId;
+  this.timesClick = 0;
+  this.timesShow = 0;
+  imageArray.push(this);
+}
+
+/**
+chooses a random index from an array
+**/
+function randomIndex(imageArray) {
+  return Math.floor(Math.random() * imageArray.length);
+}
+
+/**
+picks 3 images and returns them, no duplicates. ignores images from the previous round.
+**/
+function randomImages(remainingImages, previousImages) {
+  var chosenIndex;
+  var chosenImages = [];
+  var numToChoose = 3;
+  for (var i = 0; i < numToChoose; i++) {
+    chosenIndex = randomIndex(remainingImages);
+    chosenImages.push(remainingImages[chosenIndex]);
+    remainingImages.splice(chosenIndex, 1); // removes chosen item
```

Comment:
Avoid using splice whenever possible.  This is an incredibly inefficient operation as it requires every item following in an array to be shifted up, essentially rebuilding the array at worst case (splice from the first index).  At best it is simply a .pop() operation removing from the end.

A better approach is to record index in a separate array / object  and reference this.